frontend/assets/javascripts/src/modules/metrics.js

```diff
@@ -0,0 +1,87 @@
+/* global ga */
+define([
+    'lodash/object/extend',
+    'lodash/function/debounce',
+    'src/utils/url'
+], function (extend, debounce, urlHelper) {
+
+    var METRIC_ATTRS = {
+        'trigger': 'data-metric-trigger',
+        'category': 'data-metric-category',
+        'action': 'data-metric-action',
+        'label': 'data-metric-label' // Optional
+    };
+
+    function init() {
+        var trackingElems = document.querySelectorAll('[' + METRIC_ATTRS.trigger + ']');
+        if (window.ga && trackingElems) {
+            configureListeners(trackingElems);
+        }
+    }
+
+    function logMetric(options) {
+        var event = false;
+        if (window.ga) {
+            event = ga('send', extend({
+                'hitType': 'event'
+            }, options));
+        }
+        return event;
+    }
+
+    function configureListeners(trackingElems) {
+        var TRACKING_EVENTS = {
+            'click' : clickListener,
+            'keyup' : keyupListener
+        };
+        [].forEach.call(trackingElems, function(elem) {
+            var key = elem.getAttribute(METRIC_ATTRS.trigger);
+            var metricEvent = TRACKING_EVENTS[key];
+            if (metricEvent) {
+                metricEvent.call(this, elem);
+            }
+        });
+    }
+
+    function extractMetricsFromListener(elem, label, extras) {
+        if (elem && label) {
+            logMetric(extend({
```

Comment:
In here, would it be worth checking for the existence of all the required attributes before logging the event? Just in case someone forgets/misspells one and we lose data?